src/main/java/com/elytradev/concrete/block/ConcreteBlock.java

```diff
@@ -0,0 +1,181 @@
+package com.elytradev.concrete.block;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.SoundType;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.item.Item;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+
+import java.util.Optional;
+import java.util.Random;
+import java.util.function.Supplier;
+
+/**
+ * An extension of {@link Block} used by all blocks in Unnamed.
+ */
+public class ConcreteBlock extends Block {
+
+    protected final Supplier<Item> dropped;
+    protected final ItemDropBehaviour itemDropBehaviour;
+    protected final ExpDropBehaviour expDropBehaviour;
+
+    public ConcreteBlock(String identifier, Material materialIn, Supplier<Item> dropped,
+            ItemDropBehaviour itemDropBehaviour, ExpDropBehaviour expDropBehaviour) {
+        super(materialIn);
+        this.dropped = dropped;
+        this.itemDropBehaviour = itemDropBehaviour;
+        this.expDropBehaviour = expDropBehaviour;
+
+        this.setRegistryName(identifier);
+    }
+
+    @Override
+    public Item getItemDropped(IBlockState state, Random rand, int fortune) {
+        if (this.dropped != null) {
+            return this.dropped.get();
+        }
+        return super.getItemDropped(state, rand, fortune);
+    }
+
+    @Override
+    public int quantityDropped(Random random) {
+        return this.itemDropBehaviour.getQuantityDropped(random);
+    }
+
+    @Override
+    public int quantityDroppedWithBonus(int fortune, Random random) {
+        return this.itemDropBehaviour.getQuantityDroppedWithBonus(fortune, random);
+    }
+
+    @Override
+    public int getExpDrop(IBlockState state, IBlockAccess world, BlockPos pos, int fortune) {
+        return this.expDropBehaviour.getQuantityDropped(state, world, pos, fortune);
+    }
+
+    @Override // this is needed for its public access modifier
+    public Block setSoundType(SoundType sound) {
+        return super.setSoundType(sound);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder oreBuilder() {
+        return new Builder()
+                .drop(ItemDropBehaviour.Ore.DEFAULT);
+    }
+
+    public static Builder glassBuilder() {
```

Comment:
Why are there two specialized builders? Maybe there should be a `preset` method on the builder that takes a `Consumer<Builder>`?